charmhelpers/contrib/charmsupport/nrpe.py

```diff
@@ -313,7 +324,23 @@ def write(self):
         monitor_ids = relation_ids("local-monitors") + \
             relation_ids("nrpe-external-master")
         for rid in monitor_ids:
-            relation_set(relation_id=rid, monitors=yaml.dump(monitors))
+            reldata = relation_get(unit=local_unit(), rid=rid)
+            if 'monitors' in reldata:
+                # update the existing set of monitors with the new data
+                old_monitors = yaml.safe_load(reldata['monitors'])
+                old_nrpe_monitors = old_monitors['monitors']['remote']['nrpe']
+                # remove keys that are in the remove_check_queue
+                old_nrpe_monitors = {k: v for k, v in old_nrpe_monitors.items() if k not in self.remove_check_queue}
```

Comment:
This line should fail PEP E501 line too long.  if you reformat it to:

```python
old_nrpe_monitors = {k: v for k, v in old_nrpe_monitors.items()
                     if k not in self.remove_check_queue}
```

it'll look nicer too.